test/visualize/test_visualize.py

```diff
@@ -109,10 +109,9 @@ def create_optimization_result():
             grad=np.array([2.5 + j + 0.1, 2 + j + 1]),
         )
         result.optimize_result.append(optimizer_result=optimizer_result)
-    for j in range(0, 4):
+    for j in range(0, n):
         optimizer_result = pypesto.OptimizerResult(
-            id=str(j + 3),
-            fval=10 + j * 0.01,
+            id=str(j + 3), fval=10 + j * 0.01,
```

Comment:
Might be interpreted as a complex number (i.e. might be better to use a different name for the index, instead of `j`).